src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessedEnvelopeNotificationHandler.java

```diff
@@ -69,39 +71,44 @@ public void notifyException(Throwable throwable, ExceptionPhase exceptionPhase)
         );
     }
 
-    private void tryFinaliseMessage(IMessage message, MessageProcessingResult processingResult) {
-        try {
-            finaliseMessage(message, processingResult);
-        } catch (Exception e) {
-            log.error(
-                "An error occurred when trying to finalise 'processed envelope' message with ID {}",
-                message.getMessageId(),
-                e
-            );
-        }
+    private CompletableFuture<Void> tryFinaliseMessageAync(
+        IMessage message,
+        MessageProcessingResult processingResult
+    ) {
+        return finaliseMessageAsync(message, processingResult)
+            .exceptionally(error -> {
+                log.error(
+                    "An error occurred when trying to finalise 'processed envelope' message with ID {}",
+                    message.getMessageId(),
+                    error
+                );
+
+                return null;
+            });
```

Comment:
not sure this falls into `handleAsync` in main method. do you have integration test for that?

Reply:
This won't fall into `handleAsync` in main method. The error won't be propagated further, which is what we want.

Reply:
then why the need of a `handelAync`?

Reply:
I'm being extra cautious. In the past we've had situations where exceptions were supposed to be handled at a lower level, but it turned out that some error cases weren't covered, which resulted with failures with no logs. All you could see was that it was going fine, but never completed.